src/data/FilterCondition.py

```diff
@@ -0,0 +1,119 @@
+import shlex
+from MetaDataItem import MetaDataItem
+
+def is_string(token):
+    return (token[0] == '"' and token[-1] == '"') or (token[0] == "'" and token[-1] == "'")
+
+def is_number(token):
+    try:
+        float(token)
+    except ValueError:
+        return False
+    return True
+        
+class FilterCondition(object):
+    OPERATORS = {
+        '==',
+        '!=',
+        '>',
+        '>=',
+        '<',
+        '<=',
+        'and',
+        'or',
+        '(',
+        ')'
+    }
+    
+    VALID_ATTRS = MetaDataItem.attributes()
```

Comment:
Unlike Java, we don't need a static function to initialize a static variable. So we could do this directly
```suggestion
VALID_ATTRS = {
            'title': str,
            'url' : str,
            'collision_type' : str,
            'description' : str,
            'location' : str
        }
```

Reply:
We would need this attributes function in the FirebaseAccessor class as well, so I believe there is enough reason to maintain this static function.

Reply:
The attributes function in MetaDataItem could be converted to a static variable instead while still being usable by other classes, but having it be a function does provide one benefit. Other classes can't modify MetaDataItem's attribute list if it's returned this way. They can change their own copy, but it won't change what gets returned by attributes().